### PR TITLE
ci: update mac ci setup

### PR DESCRIPTION
**Pull request by junr03**
Description: move mac ci setup to use envoy's script. I don't see much of a point in keeping a separate script here. For instance, https://github.com/envoyproxy/envoy/pull/10357 updated to fix bazel after CI images broke. Instead of keeping up-to-date we can piggy back on efforts in upstream Envoy. Obviously, if diverging needs evolve we might need a second script in Envoy Mobile that calls the Envoy script, and does additional work.
Risk Level: low, CI will let us know if I screwed up.
Testing: CI

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

File: ci/mac_ci_setup.sh [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-fi
-
-DEPS="automake cmake coreutils go libtool wget ninja gpg bazelbuild/tap/bazelisk"
-for DEP in ${DEPS}
-do
-    is_installed "${DEP}" || install "${DEP}"
-done
-
-# https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/macos/macos-10.14-Readme.md#xcode
-sudo xcode-select --switch /Applications/Xcode_11.3.app
```
Comment (rebello95) [inline ci/mac_ci_setup.sh]: Actually this will probably break CI

**Comment by rebello95** [comment]
Is the submodule update deliberate?

File: ci/mac_ci_setup.sh (hunk 1 above)
Comment (junr03) [reply ci/mac_ci_setup.sh]: the 10.15 machines have 11.3.1 and I updated `.bazelrc`

**Comment by junr03** [comment]
Yep, because we need https://github.com/envoyproxy/envoy/pull/10357

**Pull request merged**